sgv2-docsapi/src/main/java/io/stargate/sgv2/docsapi/api/v2/namespaces/collections/jsonschema/JsonSchemaResource.java

```diff
@@ -0,0 +1,147 @@
+import io.stargate.sgv2.docsapi.config.constants.OpenApiConstants;
+import io.stargate.sgv2.docsapi.service.schema.JsonSchemaManager;
+import io.stargate.sgv2.docsapi.service.schema.TableManager;
+import io.stargate.sgv2.docsapi.service.schema.qualifier.Authorized;
+import javax.inject.Inject;
+import javax.validation.constraints.NotNull;
+import javax.ws.rs.Consumes;
+import javax.ws.rs.GET;
+import javax.ws.rs.PUT;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.UriInfo;
+import org.eclipse.microprofile.openapi.annotations.Operation;
+import org.eclipse.microprofile.openapi.annotations.media.Content;
+import org.eclipse.microprofile.openapi.annotations.media.ExampleObject;
+import org.eclipse.microprofile.openapi.annotations.parameters.Parameter;
+import org.eclipse.microprofile.openapi.annotations.parameters.Parameters;
+import org.eclipse.microprofile.openapi.annotations.responses.APIResponse;
+import org.eclipse.microprofile.openapi.annotations.responses.APIResponses;
+import org.eclipse.microprofile.openapi.annotations.security.SecurityRequirement;
+import org.eclipse.microprofile.openapi.annotations.tags.Tag;
+import org.jboss.resteasy.reactive.RestResponse;
+
+/** JSON Schema resource. */
+@Path(JsonSchemaResource.BASE_PATH)
+@Produces(MediaType.APPLICATION_JSON)
+@Consumes(MediaType.APPLICATION_JSON)
+@SecurityRequirement(name = OpenApiConstants.SecuritySchemes.TOKEN)
+@Tag(ref = OpenApiConstants.Tags.JSON_SCHEMAS)
+public class JsonSchemaResource {
+
+  public static final String BASE_PATH =
+      "/v2/namespaces/{namespace:\\w+}/collections/{collection:\\w+}/json-schema";
+
+  @Inject @Authorized TableManager tableManager;
+
+  @Inject JsonSchemaManager jsonSchemaManager;
+
+  @Operation(
+      description =
+          "Assign a JSON schema to a collection. This will erase any schema that already exists.")
+  @Parameters(
+      value = {
+        @Parameter(
+            name = "namespace",
+            ref = OpenApiConstants.Parameters.NAMESPACE,
+            description = "The namespace of the collection"),
+        @Parameter(
+            name = "collection",
+            ref = OpenApiConstants.Parameters.COLLECTION,
+            description = "The collection to assign a JSON schema to.")
+      })
+  @APIResponses(
+      value = {
+        @APIResponse(responseCode = "200", description = "Call successful."),
+        @APIResponse(
+            responseCode = "400",
+            description = "Bad request.",
+            content =
+                @Content(
+                    examples = {@ExampleObject(ref = "Invalid JSON schema.")},
+                    schema =
+                        @org.eclipse.microprofile.openapi.annotations.media.Schema(
+                            implementation = ApiError.class))),
+        @APIResponse(ref = OpenApiConstants.Responses.GENERAL_400),
```

Comment:
so this does not work, you can not have two `@APIResponse` with the same response code.. I guess this should go out?